apps/dav/lib/DAV/ViewOnlyPlugin.php

```diff
@@ -70,11 +71,16 @@ public function checkViewOnly(RequestInterface $request): bool {
 		try {
 			assert($this->server !== null);
 			$davNode = $this->server->tree->getNodeForPath($path);
-			if (!($davNode instanceof DavFile)) {
+			if (!($davNode instanceof DavFile) && !($davNode instanceof VersionFile)) {
 				return true;
 			}
-			// Restrict view-only to nodes which are shared
-			$node = $davNode->getNode();
+
+			if ($davNode instanceof DavFile) {
+				// Restrict view-only to nodes which are shared
+				$node = $davNode->getNode();
+			} else {
+				$node = $davNode->getVersion()->getSourceFile();
+			}
```

Comment:
```suggestion
			} else if ($davNode instanceof VersionFile) {
				$node = $davNode->getVersion()->getSourceFile();
			} else {
				return true;
			}
```

Then you can drop the first condition and access to `VersionFile` methods is more explicit.

Reply:
ok, I'll go with that. thanks a lot!